compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -394,6 +395,27 @@ fn inner_mir_for_ctfe(tcx: TyCtxt<'_>, def: LocalDefId) -> Body<'_> {
     body
 }
 
+fn const_prop_lint(tcx: TyCtxt<'_>, def: LocalDefId) {
```

Comment:
Can we use this as an opportunity to use a better name? `lint_arithmetic_overflow` ?